lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -84,24 +84,37 @@ defmodule ExUnit.CaptureLog do
   end
 
   defp add_capture(pid, opts) do
-    GenEvent.add_mon_handler(Logger, {Console, pid}, {pid, opts})
+    case :proc_lib.start(__MODULE__, :init_proxy, [pid, opts, self()]) do
+      :ok ->
+        :ok
+      other ->
+        mfa = {ExUnit.CaptureLog, :add_capture, [pid, opts]}
+        exit({other, mfa})
+    end
   end
 
-  defp remove_capture(pid) do
-    case GenEvent.remove_handler(Logger, {Console, pid}, nil) do
+  @doc false
+  def init_proxy(pid, opts, parent) do
+    case :gen_event.add_sup_handler(Logger, {Console, pid}, {pid, opts}) do
       :ok ->
+        ref = Process.monitor(parent)
+        :proc_lib.init_ack(:ok)
         receive do
+          {:DOWN, ^ref, :process, ^parent, _reason} -> :ok
           {:gen_event_EXIT, {Console, ^pid}, _reason} -> :ok
         end
-      {:error, :not_found} = error ->
+      other ->
+        :proc_lib.init_ack(other)
+    end
+  end
+
+  defp remove_capture(pid) do
+    case :gen_event.delete_handler(Logger, {Console, pid}, :shutdown) do
```

Comment:
This `:shutdown` could be confusing since third argument is not a reason but an arbitrary argument, I think it'd be better to stick with `:ok` which we used previously in other places.